classes/omnect_uboot_embedded_version.bbclass

```diff
@@ -0,0 +1,206 @@
+# This class implements a function to embed u-boot version in bootloader
+# binary.
+#
+# Actually it is the other way round: the built binary gets extended to place
+# the version information into the newly allocated space at the configured
+# offset.
+#
+# There are two cases of bootloader binaries:
+#  - binary blob directly written to flash space at a certain offset
+#    (common U-Boot case)
+#  - binary file in boot filesystem
+#    (special Raspberry Pi case where 2nd stage bootloader resides in
+#     kernel8.img)
+#
+# The bootloader version itself doesn't get computed here but in class
+# omnect_bootloader_versioning.bbclass which also writes the computed version
+# to file omnect_bootloader_version in WORKDIR.
+#
+# The following bitbake variables control how the version information gets
+# embedded:
+#  - OMNECT_BOOTLOADER_EMBEDDED_VERSION_BBTARGET
+#    defines the bitbake target the versioned bootloader binary bases upon
+#  - OMNECT_BOOTLOADER_EMBEDDED_VERSION_BBFILE
```

Comment:
`_BBFILE` irritates: it's not a `bb` file and `base binary` is not really intuitive to me.
`_BINFILE` ?

Reply:
right, the name was a bad choice, renamed to BINFILE.